libcnb/src/util.rs

```diff
@@ -12,10 +16,50 @@ pub(crate) fn default_on_not_found<T: Default>(
     }
 }
 
+/// Recursively removes the given path, similar to [`std::fs::remove_dir_all`].
+///
+/// Compared to `remove_dir_all`, this function behaves more like `rm -rf` on UNIX systems. It will
+/// delete files and directories even if their permissions would normally prevent deletion as long
+/// as the current user is the owner of these files (or root).
+pub(crate) fn remove_recursively<P: AsRef<Path>>(path: P) -> std::io::Result<()> {
+    if path.as_ref().symlink_metadata()?.is_dir() {
+        // To delete a directory, the current user must have the permission to write and list the
+        // directory (to empty it before deleting). To reduce the possibility of permission errors,
+        // we try to set the correct permissions before attempting to delete the directory and the
+        // files within it.
+        let permissions = if cfg!(target_family = "unix") {
+            use std::os::unix::fs::PermissionsExt;
+            Permissions::from_mode(0o777)
+        } else {
+            let mut permissions = path.as_ref().metadata()?.permissions();
+            permissions.set_readonly(false);
+            permissions
+        };
+
+        fs::set_permissions(&path, permissions)?;
+
+        for entry in fs::read_dir(&path)? {
+            let path = entry?.path();
+
+            // Since the directory structure could be a deep, blowing the stack is a real danger
+            // here. We use the `stacker` crate to allocate stack on the heap if we're below
+            // 4kb of remaining stack.
+            stacker::maybe_grow(4096, 32768, || remove_recursively(&path))?;
```

Comment:
These are indeed magic numbers without explanation. 

The stack crate itself does not document a sensible default value or any other guidance how to choose these values. In the end, `4kb` seemed enough to run one layer of `remove_recursively` and `32kb` didn't seem to be too much or too little. Nothing of this is backed by any data.

Adding a comment that these are numbers that seemed fine but there isn't any particular reason they've been choose  is something we can do. It would certainly make it easier for future generations to decide to change the value.

Adding constants for these too values seems excessive when they're only used in one place (right now) and their values aren't backed by data. They're very much local "best guesses".

Reply:
yeah, that's why i left it as "descriptive names and/or rational commented" 😅   

was just looking for some extra context so that anyone coming back to this code will know if they were chosen for a specific reason or they are arbitrary.